File: leap.cpp
#include<iostream>
using namespace std;
int main()
{
    int year;
    cout<<"enter a year"<<endl;
    cin>>year;
    if(year%4==0 and year%100!=0 or year%400==0)
    {
        cout<<"year  "<<year<<" is leap"<<endl;
    }
    else{
        cout<<"year "<<year<<" is not leap"<<endl;
    }
}

File: leap.java
import java.util.Scanner;
class leap{
    public static void main(String args[])
    {
        int year;
        Scanner input=new Scanner(System.in);
        System.out.println("enetr any year");
        year=input.nextInt();
        if(year%4==0 & year%100!=0 ^ year%400==0)
        {
            System.out.println("the year " +year+ " is leap year");
        }
        else{
            System.out.println("the year "  +year+ " is  not leap year");
        }



    }
}
